### PR TITLE
Added: tests on auto-stop. Refactored: auto-stop lib

### PR DESCRIPTION
**Pull request by jsilvestre**
~~It's still work in progress, there is still the integration test to write.~~ It's all good.

### CODE REVIEW

**Comment by jsilvestre** [comment]
@m4dz I am done with this PR. As usual, all tests pass but Travis is not happy for some reason.

**Comment by jsilvestre** [comment]
Today we celebrate the green status of the Home Travis build!

**Comment by frankrousseau** [comment]
\o/

**Comment by jsilvestre** [comment]
I removed a piece of code that was used in the past. I checked with @poupotte that it was okay to remove it. Our changes in the way the Data System, the Controller and the Home interacts made this code obsolete.

File: Cakefile [hunk 1]
```diff
@@ -45,15 +45,18 @@ task 'tests', "Run tests #{taskDetails}", (opts) ->
...
     command += "--compilers coffee:coffee-script/register"
     exec command, (err, stdout, stderr) ->
         console.log stdout
-        if err or stderr
-            console.log stderr
-            console.log "Running mocha caught exception:\n" + err
-            setTimeout (-> process.exit 1), 100
+        if err?
+            console.log "Running mocha caught exception:\n#{err}"
+            process.exit 1
```
Comment (aenario) [inline Cakefile]: you should setTimeout before process.exit, otherwise the console.log is lost sometimes

File: server/lib/autostop.coffee [hunk 2]
```diff
@@ -20,51 +29,61 @@ Stop application <app> :
...
-                        app.state is "installed"
-                            console.log "stop : " + name
-                            stopApp app
-    ,5 * 60 * 1000
+module.exports.startTimeout = (slug) ->
+    unless slug in ['home', 'proxy']
+        applicationTimeout[slug] = setTimeout ->
+            Application.all key: slug, (err, apps) ->
+                return log.error err if err?
+                return log.error "App #{slug} not found" unless apps?.length > 0
```
Comment (aenario) [inline server/lib/autostop.coffee]: > 80 ?

Comment (jsilvestre) [inline server/lib/autostop.coffee]: it's exactly 80 :dancer:

File: test/autostop.coffee [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+                should.not.exist err
+                should.exist apps
+                apps[0].state.should.equal 'installed'
+
+        it "Then I wait for one more minute", ->
+            @sandbox.clock.tick 2 * minute
+            @sandbox.clock.restore()
+
+        it "Then the application should be stopped", (done) ->
+            @timeout 6000
```
Comment (aenario) [inline test/autostop.coffee]: I find it cleaner to use separate test for waiting
```coffee
it "Then I wait a bit", (done) ->
   @timeout 6000
   setTimeout done 5000

...
```

Not sure if it still necessary / incompatible with the sandbox.clock

Comment (jsilvestre) [inline test/autostop.coffee]: I do the change. It's okay to use a timeout because I restore the setTimeout mocking in the previous step.

**Comment by aenario** [comment]
OK, everything LGTM

**Comment by aenario** [comment]
oups, didn't notice #275 was piggy backing :)
Updated on my cozy, this is not broken.

**Pull request merged**